archive/walk.go

```diff
@@ -2,18 +2,235 @@ package archive
 
 import (
 	"context"
+	"fmt"
 	"io"
+	"path"
+	"regexp"
+	"sort"
+	"strconv"
 	"strings"
+	"time"
 
 	"github.com/brimsec/zq/driver"
 	"github.com/brimsec/zq/pkg/iosrc"
+	"github.com/brimsec/zq/pkg/nano"
 	"github.com/brimsec/zq/scanner"
 	"github.com/brimsec/zq/zbuf"
 	"github.com/brimsec/zq/zio"
 	"github.com/brimsec/zq/zio/detector"
 	"github.com/brimsec/zq/zng/resolver"
+	"github.com/segmentio/ksuid"
 )
 
+const (
+	dataDirname = "zd"
+	zarExt      = ".zar"
+)
+
+// A fileKind is the first part of a file name, used to differentiate files
+// when they are listed from the archive's backing store.
+type fileKind string
+
+const (
+	unknownFileKind fileKind = ""
+	ingestDataKind           = "d"
+	seekIndexKind            = "ts"
+)
+
+// A uuid is a unique identifier created with the ksuid package. These ids
+// are always 27 characters long, and combine 32 bits of time plus 128
+// random bits. The ids are roughly sortable by generation time, which can
+// be useful for debugging & investigation.
+type uuid string
+
+func newUUID() uuid {
+	return uuid(ksuid.New().String())
+}
```

Comment:
Why not just use github.com/segmentio/ksuid.KSUID for this?

Reply:
👍 I thought about using something other than jus ksuid.String(), like adding some prefix (like we do with space id's). But file names will have a prefix already (like "d-..." for data files) that will help explain what the id represents.